src/CoreBundle/EventListener/AllowEmbeddingForDifferentDomainListener.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Chameleon System (https://www.chameleonsystem.com).
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ChameleonSystem\CoreBundle\EventListener;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+/**
+ * Allow the calling domain to see this page (e.g. in an iframe).
+ */
+class AllowEmbeddingForDifferentDomainListener
+{
+    /**
+     * @param GetResponseEvent $event
+     */
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+
+        if (true === $this->canAllowAdditionalDomains($request)) {
```

Comment:
why "can" and not just `allowAdditionalDomains`?

Reply:
i would also prefer early exit here to keep the method simple.

Reply:
ok